src/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryMetadataEntry.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Identity.Client.Instance.Discovery
+{
+    [DataContract]
+    internal sealed class InstanceDiscoveryMetadataEntry
+    {
+        [DataMember(Name = "preferred_network")]
+        public string PreferredNetwork { get; set; }
+
+        [DataMember(Name = "preferred_cache")]
+        public string PreferredCache { get; set; }
+
+        [DataMember(Name = "aliases")]
+        public string[] Aliases { get; set; }
+
+        public IEnumerable<string> AliasesWithPreferredCacheFirst
```

Comment:
rename to verb:  GetAliasesWithPreferredCacheFirst().  Also, make this a method instead of a property since it's doing some actual work.